cdap-tms/src/main/java/co/cask/cdap/messaging/store/AbstractMessageTable.java

```diff
@@ -158,6 +164,16 @@ protected Entry computeNext() {
 
       while (scanner.hasNext()) {
         RawMessageTableEntry tableEntry = scanner.next();
+
+        // See if we need to skip the first row returned by the scanner
+        if (skipStartRow != null) {
```

Comment:
Do we need a similar change in AbstractPayloadTable - ie, to compare and skip startRow, if there is a match, instead of skipping first Row?